server/util/csv_parse.py

```diff
@@ -61,14 +68,51 @@ def parse_csv(csv_string: str, columns: List[CSVColumnType]) -> CSVDictIterator:
     return dict_csv
 
 
-def validate_is_csv(csv: str):
-    lines = csv.splitlines()
-    if len(lines) == 0:
+def validate_csv_mimetype(file: FileStorage) -> None:
+    # In Windows, CSVs have mimetype application/vnd.ms-excel
+    if file.mimetype not in ["text/csv", "application/vnd.ms-excel"]:
+        raise BadRequest(INVALID_CSV_ERROR)
+
+
+def decode_csv(file: BinaryIO) -> TextIO:
+    detector = chardet.UniversalDetector()
+    for i, line in enumerate(file.readlines()):
+        detector.feed(line)
+        if detector.done or i > 500:
+            break
+    detector.close()
+    if not detector.result["encoding"]:
+        raise CSVParseError(INVALID_CSV_ERROR)
+    encoding = detector.result["encoding"]
+    if encoding == "ascii":
+        encoding = "utf-8"
+    file.seek(0)
+    return io.TextIOWrapper(file, encoding=encoding, newline=None)
+
+
+# TODO remove this once we decode all CSV files using decode_csv
+def decode_csv_file(file: FileStorage) -> str:
+    try:
+        contents = file.read()
+        if contents == b"":
+            return ""
+        return decode_csv(io.BytesIO(contents)).read()
+    except CSVParseError as err:  # pragma: no cover
+        raise BadRequest(INVALID_CSV_ERROR) from err
+
+
+def validate_not_empty(file: BinaryIO):
```

Comment:
Should this function be responsible for seeking back to 0?

Reply:
oh definitely! i intended to do that and forgot. i'm actually kind of shocked this didn't break everything...

Reply:
Ah, I think I know why it didn't break - because `validate_comma_delimited` later seeks back to 0. That was just luck though, I will fix